test-site/src/pages/LocationsPage.tsx

```diff
@@ -0,0 +1,76 @@
+import { useSearchActions, Matcher, useSearchState, SelectableStaticFilter } from '@yext/search-headless-react';
+import {
+  AppliedFilters,
+  ResultsCount,
+  SearchBar,
+  StandardCard,
+  VerticalResults,
+  LocationBias,
+  MapboxMap,
+  Pagination,
+  MapboxMapProps,
+} from '@yext/search-ui-react';
+import { LngLat, LngLatBounds } from 'mapbox-gl';
+import { useCallback, useLayoutEffect } from 'react';
+import { MapPin } from '../components/MapPin';
+
+export interface Location {
+  yextDisplayCoordinate: {
+    latitude: number,
+    longitude: number
+  } | undefined
+}
+
+const mapboxOptions: MapboxMapProps<Location>['mapboxOptions'] = {
+  zoom: 10
+};
+
+export function LocationsPage() {
+  const searchActions = useSearchActions();
+  const filters = useSearchState(state => state.filters.static);
+  useLayoutEffect(() => {
+    searchActions.setVertical('KM');
+    searchActions.executeVerticalQuery();
+  }, [searchActions]);
+
+  const onDrag: MapboxMapProps<Location>['onDrag'] = useCallback(
+    (center: LngLat, bounds: LngLatBounds) => {
+      const radius = center.distanceTo(bounds.getNorthEast());
+      const nonLocationFilters: SelectableStaticFilter[] = filters?.filter(f => f.filter.kind !== 'fieldValue' || f.filter.fieldId !== 'builtin.location') ?? [];
+      const nearFilter: SelectableStaticFilter = {
```

Comment:
nit: do we want to give this filter some kind of generic displayName so applied filters doesn't just display it as a blank pill with no text?

Reply:
sure thing, update to have 'Near Current Area' as the display name